cli/compose/convert/volume.go

```diff
@@ -88,3 +67,77 @@ func convertVolumeToMount(
 	// Named volumes
 	return result, nil
 }
+
+func handleBindToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type bind")
+	}
+	if volume.Tmpfs != nil {
+		return result, errors.New("tmpfs options are incompatible with type bind")
+	}
+	if volume.Bind != nil {
+		result.BindOptions = &mount.BindOptions{
+			Propagation: mount.Propagation(volume.Bind.Propagation),
+		}
+	}
+	// Binds volumes
+	return result, nil
+}
+
+func handleTmpfsToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	// source must be empty for tmpfs
+	if volume.Source != "" {
+		return result, errors.New("invalid tmpfs source, source must be empty")
+	}
+	if volume.Bind != nil {
+		return result, errors.New("bind options are incompatible with type tmpfs")
+	}
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type tmpfs")
+	}
+	if volume.Tmpfs != nil {
+		result.TmpfsOptions = &mount.TmpfsOptions{
+			SizeBytes: volume.Tmpfs.Size,
+		}
+	}
+	// Tmpfs volumes
```

Comment:
This comment doesn't add much, so I'd remove it; if the function needs documentation, add a GoDoc for the function itself